### PR TITLE
Added build phases script for running SwiftLint at build time

### PR DESCRIPTION
**Pull request by addb**
Added script in build phases to perform swift lint while building the project in Xcode. Also cleaned up few duplicate compile sources from the project.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Tests/UnitTests/MediaContextTests.swift [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import AEPCore
+@testable import AEPEdgeMedia
+import XCTest
+
+class MediaContextTests: XCTestCase {
+    let mediaInfo = MediaInfo(id: "mediaID", name: "mediaName", streamType: "aod", mediaType: MediaType.Audio, length: 30.0, prerollWaitingTime: 0)!
```
Comment (emdobrin) [inline Tests/UnitTests/MediaContextTests.swift:18]: should we be using the public constants for the test inputs here, e.g. StreamType.aod

File: Tests/UnitTests/MediaContextTests.swift [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    let mediaMetadata = ["media.show": "sampleshow", "key1": "value1", "key2": "мểŧẳđαţả"]
+    let adBreakInfo = AdBreakInfo(name: "adBreakName", position: 1, startTime: 1.1)!
+    let adInfo = AdInfo(id: "adID", name: "adName", position: 1, length: 15.0)!
+    let adMetadata = ["media.ad.advertiser": "sampleAdvertiser", "key1": "value1", "key2": "мểŧẳđαţả"]
```
Comment (emdobrin) [inline Tests/UnitTests/MediaContextTests.swift:22]: Can we use the public constants for the main flow, eg. AdMetadataKeys.ADVERTISER,  AudioMetadataKeys.ARTIST?
We can still have tests with custom keys, but maybe add a separate test just for that

File: Tests/UnitTests/MediaContextTests.swift [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        XCTAssertTrue(mediaContext.isInMediaPlaybackState(state: MediaContext.MediaPlaybackState.Play))
+        XCTAssertFalse(mediaContext.isIdle())
+
+        mediaContext.enterPlaybackState(state: MediaContext.MediaPlaybackState.Pause)
+        XCTAssertTrue(mediaContext.isInMediaPlaybackState(state: MediaContext.MediaPlaybackState.Pause))
+        XCTAssertTrue(mediaContext.isIdle())
+
+        mediaContext.enterPlaybackState(state: MediaContext.MediaPlaybackState.Seek)
+        XCTAssertTrue(mediaContext.isInMediaPlaybackState(state: MediaContext.MediaPlaybackState.Seek))
+        XCTAssertTrue(mediaContext.isIdle())
```
Comment (emdobrin) [inline Tests/UnitTests/MediaContextTests.swift:191]: Can we have a separate test just for testing the behavior for idle?

File: Tests/UnitTests/MediaContextTests.swift [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+    func testGetAllStates_returnsAllStatesWithTheirCurrentStatus() {
+        let mediaContext = MediaContext(mediaInfo: mediaInfo, metadata: mediaMetadata)
+        for stateName in 1...10 {
+            XCTAssertTrue(mediaContext.startState(info: StateInfo(stateName: "State\(stateName)")!))
+        }
+
+        var activeStatesList = mediaContext.getActiveTrackedStates()
+        XCTAssertEqual(10, activeStatesList.count)
+        for stateName in 1...10 {
+            XCTAssertTrue(activeStatesList.contains(StateInfo(stateName: "State\(stateName)")!))  // 1-5 states are active
```
Comment (emdobrin) [inline Tests/UnitTests/MediaContextTests.swift:161]: this comment belongs to L171, here seems like all states should be active after they were started

File: Tests/UnitTests/MediaContextTests.swift (hunk 1 above)
Comment (emdobrin) [reply Tests/UnitTests/MediaContextTests.swift:18]: you left this one out. the other changes look good 👍

**Review by emdobrin — APPROVED** [review]
Looks good to me

**Review by kevinlind — APPROVED** [review]

**Pull request merged**